### ISSUE TITLE
Google sgin in platform exceptions

### ISSUE THREAD

**Issue by AhmedAlaa96** [flutter/flutter#56235]
Hello, everyone

I'm trying to build Continue with Google and I'm using `google_sign_in` 
Everything is fine when I build in my laptop but when I try to build in another laptop give me `PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)` and when I searched for this error I found there is misconfigurations in `google-services.json` `SHA-1` and `SHA-256` and there is sometimes I got `platformexception(sign_in_failed, com.google.android.gms.common.api.apiexception: 12500: , null)` when I searched for it I found steps to solve it and check my google services on my device and still the same 
What should I do because when I release the application for testing in app center give me one of those errors and I don't understand why 

Thank you

**Comment by fr3d3rico** [comment]
I had been it for a while!
Today I found a way to handle it.
After read this discussion:
[https://github.com/flutter/friendlychat-steps/issues/27][1]

Take a look at this answer:

> In Step 5, the signing key is the debug signing certificate (SHA-1)
> described in Configure Firebase for Android. You retrieved it from
> your development machine and copied it to the dialog in Firebase
> Console.
> 
> This key has to be from the machine you're developing on. If you have
> both a workstation and a laptop, make sure it matches the one where
> you're running friendlychat.

Well, I search for a new way to generate SHA-1 and reach this discussion:
[https://stackoverflow.com/questions/51845559/generate-sha-1-for-flutter-app][2]

Try it! I hope it help you.


  [1]: https://github.com/flutter/friendlychat-steps/issues/27
  [2]: https://stackoverflow.com/questions/51845559/generate-sha-1-for-flutter-app

**Comment by iapicca** [comment]
Hi @AhmedAlaa96 
does the proposed [solution](https://github.com/flutter/flutter/issues/56235#issuecomment-630951328) works for you?
If it doesn't please provide 
your `flutter doctor -v` , 
your `flutter run --verbose`,
your `pubspec.yaml` 
and a reproducible minimal code sample.
Thank you

**Comment by DyaryRaoof** [comment]
having the same issue not solved by the above solution.

**Comment by vinodlee** [comment]
PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null)
[        ] V/PhoneWindow(11057): DecorView setVisiblity: visibility = 0, Parent = android.view.ViewRootImpl@8c0de40, this = DecorView@3565179[MainActivity]
[  +17 ms] V/PhoneWindow(11057): DecorView setVisiblity: visibility = 4, Parent = android.view.ViewRootImpl@34fb21b, this = DecorView@9b8c2cc[SignInHubActivity]
[        ] D/Surface (11057): Surface::disconnect(this=0x799192c000,api=1)
[   +7 ms] D/View    (11057): [Warning] assignParent to null: this = DecorView@9b8c2cc[SignInHubActivity]

**Comment by iapicca** [comment]
Could everyone who still has this problem please file a new issue with the information listed [above](https://github.com/flutter/flutter/issues/56235#issuecomment-631307957)
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by findsarfaraz** [comment]
I too have same issue. 

sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null

any luck with same

**Comment by WouterVandenputte** [comment]
> I too have same issue.
> 
> sign_in_failed, com.google.android.gms.common.api.ApiException: 10: , null
> 
> any luck with same

Hye, did you found a solution yet?

**Comment by teyushen** [comment]
Same question. When I build release by APK, there is no problem. But when I build on appbundle, it has same error.

> flutter build apk --split-per-abi --release --flavor prod -t lib/main_prod.dart -> it works
> flutter build appbundle --release --flavor prod -t lib/main_prod.dart  -> it has same error

**Comment by TahaTesser** [comment]
Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by marchacio** [comment]
I find the solution!

1) First of all, I list all the packages and versions that I have used (today: 17/02/2021):
**_in pubspec.yaml_**
```
firebase_auth: ^0.20.0+1
google_sign_in: ^4.5.9
firebase_core: ^0.7.0
```

**_in android/build.grandle_**
```
classpath 'com.android.tools.build:gradle:3.6.4'
classpath 'com.google.gms:google-services:4.3.5'
```

**_in android/app/build.grandle_**
```
dependencies{
   ...
   implementation 'com.google.firebase:firebase-analytics:18.0.2'
   ...
}
```

2) I solved it by adding more keys to the firebase. Initially I thought it was stupid and useless, then in the end it turned out to be the solution (I added up to 10 SHA, don't worry if that's a lot).
_Where did I find all these keys?_
I found all these keys by entering the android folder and typing the command `./gradlew signingReport` (I use linux, it also works on mac. For windows the command is `gradlew signingReport`)

3) it printed out a very long list of keys in my terminal, and I took all the ones that were different, that were at the top of the list.
_Note: you will get all the keys (both debug and release) of each package of the app, but rest assured that all those of the packages are the same and only for debugging (so you will have to take these same keys only once)_

4) Once you have all the app keys, go to your project's Firebase settings, scroll down and add any keys you found, but continue reading 'couse this issue is till not solved. So, after that, you will need to download the `google-services.json` file again and run a `flutter clean` in your project's local folder on your computer.

Now you can build your app and see if it works!

I hope this could help someone

**Comment by WouterVandenputte** [comment]
> I find the solution!
> 
> 1. First of all, I list all the packages and versions that I have used (today: 17/02/2021):
> 
> **_in pubspec.yaml_**
> 
> ```
> firebase_auth: ^0.20.0+1
> google_sign_in: ^4.5.9
> firebase_core: ^0.7.0
> ```
> 
> **_in android/build.grandle_**
> 
> ```
> dependencies {
>    ...
>    classpath 'com.android.tools.build:gradle:3.6.4'
>    classpath 'com.google.gms:google-services:4.3.5'
>    ...
> }
> ```
> 
> **_in android/app/build.grandle_**
> 
> ```
> dependencies{
>    ...
>    implementation 'com.google.firebase:firebase-analytics:18.0.2'
>    ...
> }
> ```
> 
> 1. I solved it by adding more keys to the firebase. Initially I thought it was stupid and useless, then in the end it turned out to be the solution (I added up to 10 SHA, don't worry if that's a lot).
>    _Where did I find all these keys?_
>    I found all these keys by entering the android folder and typing the command `./gradlew signingReport` (I use linux, it also works on mac. For windows the command is `gradlew signingReport`)
> 2. it printed out a very long list of keys in my terminal, and I took all the ones that were different, that were at the top of the list.
>    _Note: you will get all the keys (both debug and release) of each package of the app, but rest assured that all those of the packages are the same and only for debugging (so you will have to take these same keys only once)_
> 3. Once you have all the app keys, go to your project's Firebase settings, scroll down and add any keys you found, but continue reading 'couse this issue is till not solved. So, after that, you will need to download the `google-services.json` file again and run a `flutter clean` in your project's local folder on your computer.
> 
> Now you can build your app and see if it works!
> 
> I hope this could help someone

I solved it too in the meantime, you need not add 10 SHA keys. If you (as many probably do) set up google play signing, all you need is to add the SHA-1 AND SHA-256 key that google uses to sign your app. This can be found under `App integrity` (translation from my language, could be called something else in English)

![image](https://user-images.githubusercontent.com/23052815/108265567-3df67980-7169-11eb-9d64-45917329d51b.png)

**Comment by teyushen** [comment]
> > I find the solution!
> > 
> > 1. First of all, I list all the packages and versions that I have used (today: 17/02/2021):
> > 
> > **_in pubspec.yaml_**
> > ```
> > firebase_auth: ^0.20.0+1
> > google_sign_in: ^4.5.9
> > firebase_core: ^0.7.0
> > ```
> > 
> > 
> > **_in android/build.grandle_**
> > ```
> > dependencies {
> >    ...
> >    classpath 'com.android.tools.build:gradle:3.6.4'
> >    classpath 'com.google.gms:google-services:4.3.5'
> >    ...
> > }
> > ```
> > 
> > 
> > **_in android/app/build.grandle_**
> > ```
> > dependencies{
> >    ...
> >    implementation 'com.google.firebase:firebase-analytics:18.0.2'
> >    ...
> > }
> > ```
> > 
> > 
> > 
> > 1. I solved it by adding more keys to the firebase. Initially I thought it was stupid and useless, then in the end it turned out to be the solution (I added up to 10 SHA, don't worry if that's a lot).
> >    _Where did I find all these keys?_
> >    I found all these keys by entering the android folder and typing the command `./gradlew signingReport` (I use linux, it also works on mac. For windows the command is `gradlew signingReport`)
> > 2. it printed out a very long list of keys in my terminal, and I took all the ones that were different, that were at the top of the list.
> >    _Note: you will get all the keys (both debug and release) of each package of the app, but rest assured that all those of the packages are the same and only for debugging (so you will have to take these same keys only once)_
> > 3. Once you have all the app keys, go to your project's Firebase settings, scroll down and add any keys you found, but continue reading 'couse this issue is till not solved. So, after that, you will need to download the `google-services.json` file again and run a `flutter clean` in your project's local folder on your computer.
> > 
> > Now you can build your app and see if it works!
> > I hope this could help someone
> 
> I solved it too in the meantime, you need not add 10 SHA keys. If you (as many probably do) set up google play signing, all you need is to add the SHA-1 AND SHA-256 key that google uses to sign your app. This can be found under `App integrity` (translation from my language, could be called something else in English)
> 
> ![image](https://user-images.githubusercontent.com/23052815/108265567-3df67980-7169-11eb-9d64-45917329d51b.png)

Yes, I solved this few days ago. Same as you, just put google play's SHA-1 into firebase project. (https://keeplearning.blog/2021/02/13/flutter-signing-problem-when-building-android-app-bundle-in-release-mode/)

Thank you.

**Comment by jamilxt** [comment]
1. While adding an app on your Firebase Project, cross-check your android application's package name (e.g **com.example.application**) which you will find on **AndroidManifest.xml** or, **build.gradle** (app level) file. If the package name is wrong adding **SHA1** or **SHA512** won't be helpful. 

2. Follow the **Firebase SDK Instructions** properly. Add the required lines of code which they are recommending you to add.

3. Don't forget to add the **SHA1** or **SHA512** to your project which you can easily generate using the following command using the **terminal** of Android Studio.
 
       cd android 
       gradlew signingReport


4. Use the command `flutter clean` on the terminal before running the application.

Following the **Firebase SDK Instructions** properly will save you lots of time.

**Comment by ramonpaolo** [comment]
I had the same problem here. The solution was change the name package.
I was using other name package.
